tests/test_dataclasses.py

```diff
@@ -801,6 +801,46 @@ class Config:
         e.item.name = 'pika2'
 
 
+def test_pydantic_callable_field():
+    """pydantic callable fields behaviour should be the same as stdlib dataclass"""
+
+    def foo(arg1, arg2):
+        return arg1, arg2
+
+    @pydantic.dataclasses.dataclass
+    class HasCallablesDC:
+        non_default_callable: Callable
+        default_callable: Callable = lambda x: foo(x, 'default')
+        default_callable_factory: Callable = dataclasses.field(default=lambda x: foo(x, 'factory'))
+
+    class HasCallablesModel(BaseModel):
+        non_default_callable: Callable
```

Comment:
what about typed callables like `Callable[[int, float, str], bool]`?

Surely worth adding tests for them too?